graylog2-web-interface/src/views/components/visualizations/GenericPlot.tsx

```diff
@@ -233,7 +270,9 @@ class GenericPlot extends React.Component<GenericPlotProps, State> {
                                 layout={interactive ? plotLayout : merge({}, nonInteractiveLayout, plotLayout)}
                                 style={style}
                                 onAfterPlot={onRenderComplete}
-                                onClick={interactive ? null : () => false}
+                                onClick={interactive ? _onMarkerClick : () => false}
+                                onHover={_onHoverMarker}
+                                onUnhover={onUnhoverMarker}
```

Comment:
In this case, we always create some listeners. I think it would be better to pass some `undefined` if there is no `on${ACTION}` handler provided.

Reply:
I like the idea, but how would you do it for `onClick`, I would like to avoid `onClick={interactive ? (typeof onClickMarker === 'function' ? _onMarkerClick : undefined) : () => false} `

I also plan to refactor this part next week so we can define a custom hover label for all plots.

Reply:
@maxiadlovskii and I discussed to make this up in a follow-up PR together with further changes.